pkg/controller/container-runtime-config/helpers.go

```diff
@@ -496,15 +496,38 @@ func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec)
 	for i, reg := range imgSpec.RegistrySources.BlockedRegistries {
```

Comment:
Above we compute `payloadReg`. Do we know that all of the release is in the `releaseImage` repo? If that’s the case, we could probably make a much more accurate check, especially WRT interpreting ICSP correctly.

Reply:
Yes, all the release is in the releaseImage repo - how could we make a more accurate check here?

Reply:
Conceptually:
- Build the effective registries.conf from all the ICSP, processing overrides/merging etc. = `EditRegistriesConfig`; ideally by only calling that once in this package, and doing this registries validation on the outcome, instead of calling `EditRegistriesConfig` twice.
- Look up the effective configuration for 1) `releaseRepo` = repo of `releaseImage` 2) _any_ configuration for `releaseRepo:$anyTag` or `releaseRepo@anyDigest`. That’s basically `pkg/registries.scopeIsNestedInsideScope(entryInQuestion.{Prefix,Location}, releaseRepo`.
  - Or, alternatively, `c/image/pkg/sysregistriesv2.FindRegistry(releaseRepo)` can be used to find one matching entry, but not possible `:anyTag`/`@anyDigest` subscopes.

Reply:
We are calling `EditRegistriesConfig` only once. We are doing this validation on the initial input of blocked registries before calling EditRegistriesConfig on it.

Reply:
… actually, I was overcomplicating things. In `EditRegistries` a single `blockedScopes` entry can create multiple `sysregistriesv2.Registry` entries, but they always cover the _full_ subtree of that blocked scope; there is no way to “unblock” a sub-scope.

So, just a `ScopeIsNestedInsideScope(releaseRepo, blockedReg)` should determine if that `blockedReg` value affects `releaseRepo`. That would allow blocking `quay.io/foo` without limitation, because it doesn’t affect `releaseRepo`.

Reply:
fixed